tasks/build.yml

```diff
@@ -64,30 +65,45 @@
       or ( item.group is defined and item.group in users_and_groups.whitelist_groups)
     )
 
-- name: Create .ssh directory
+- name: Copy ssh auth keys to user home directory
+  become: yes
+  authorized_key:
+    user: "{{ item.name }}"
+    key: "{{ lookup('file', item.ssh_key) }}"
+    manage_dir: yes
+  with_items: "{{ users_and_groups.users }}"
+  when: (users_and_groups.authorized_keys_dir is not defined)
+    and (item.state is not defined or item.state != absent)
+    and (item.ssh_key is defined)
+    and (
+      (not users_and_groups.whitelist_groups)
+      or (item.groups is defined and (item.groups | intersect(users_and_groups.whitelist_groups)))
+      or (item.group is defined and item.group in users_and_groups.whitelist_groups)
+    )
+
+- name: Create authorized_keys directory
   become: yes
   file:
-    path: /home/{{ item.name }}/.ssh
-    owner: "{{ item.name }}"
-    group: "{{ item.group | default(omit) }}"
-    mode: 0700
+    path: "{{ users_and_groups.authorized_keys_dir }}"
+    owner: "root"
```

Comment:
true, if we ever need to run sshd as other user, then we can change that to a variable. For now hardcoded `root` is fine